src/pages/SignUp.tsx

```diff
@@ -1,318 +1,67 @@
-      firebaseAuth,
-      basicsEmailAddress,
-      basicsPassword
-    )
-      .then(async res => {
-        try {
-          const userId = res.user.uid;
-
-          const accountRef = await addDoc(
-            collection(firebaseFirestore, 'accounts'),
-            {
-              basics_18_plus: basics18Plus,
-              first_name: basicsFirstName,
-              last_name: basicsLastName,
-              privacy_agreement: true,
-              uid: userId
-            }
-          );
-
-          const profileRef = await addDoc(
-            collection(firebaseFirestore, 'profiles'),
-            {
-              uuid: userId,
-              account_id: accountRef.id,
-              stage_role: stageRole
-            }
-          );
-
-          setAccountRef(accountRef);
-          setProfileRef(profileRef);
-        } catch (e) {
-          console.error('Error adding document:', e);
-        }
-      })
-      .catch(err => {
-        console.log('Error creating user:', err);
-      });
-  };
-
-  // callback function for updating if a step has errors
-  // we pass this down in the "hasErrorCallback" prop for the step
-  const setStepErrorsCallback = (step: string, hasErrors: boolean) => {
-    const newStepErrorsObj = { ...stepErrors };
-
-    if (step in newStepErrorsObj) {
-      newStepErrorsObj[step] = hasErrors;
-    }
-
-    setStepErrors(newStepErrorsObj);
-  };
-
-  // based on which step we're on, return a different step component and pass it the props it needs
-  const stepFrame = () => {
-    const props = { formData, setForm, navigation };
-    let returnStep;
-
-    switch (step) {
-      case 'landing' as any:
-        returnStep = (
-          <Landing
-            {...props}
-            landingStep={landingStep}
-            setLandingStep={setLandingStep}
-          />
-        );
-        break;
-      case 'basics' as any:
-        returnStep = (
-          <Basics {...props} hasErrorCallback={setStepErrorsCallback} />
-        );
-        break;
-      case 'privacy' as any:
-        returnStep = <Privacy {...props} />;
-        break;
-      case 'actorInfo1' as any:
-        returnStep = (
-          <ActorInfo1 {...props} hasErrorCallback={setStepErrorsCallback} />
-        );
-        break;
-      case 'actorInfo2' as any:
-        returnStep = (
-          <ActorInfo2 {...props} hasErrorCallback={setStepErrorsCallback} />
-        );
-        break;
-      case 'offstageRoles' as any:
-        returnStep = <OffstageRoles {...props} />;
-        break;
-      case 'profilePhoto' as any:
-        returnStep = <ProfilePhoto {...props} />;
-        break;
-      case 'demographics' as any:
-        returnStep = <Demographics {...props} />;
-        break;
-      case 'profilePreview' as any:
-        returnStep = <Profile previewMode={true} />;
-        break;
-      default:
-        returnStep = <></>;
-        break;
-    }
+  }
+
+  if (accountType === 'individual') {
```

Comment:
[nit] this may look better as a switch/case as well

Reply:
Yeah I will take a look at that too.